sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AbstractKafkaPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,185 @@
+	 * @param properties the Kafka Spring properties
+	 * @return the map from Kafka properties storing producer-specific properties
+	 */
+	protected abstract Map<String, String> getRawProducerProperties(T properties);
+
+	/**
+	 * Create a map of the merged Kafka consumer properties from the Kafka Spring properties.
+	 * @param properties the Kafka Spring properties
+	 * @return a Map containing all Kafka consumer properties
+	 */
+	protected abstract Map<String, Object> getMergedConsumerProperties(T properties);
+
+	/**
+	 * Get the raw {@link Map} object from the Kafka Spring properties that stores consumer-specific properties.
+	 * @param properties the Kafka Spring properties
+	 * @return the map from Kafka properties storing consumer-specific properties
+	 */
+	protected abstract Map<String, String> getRawConsumerProperties(T properties);
+
+	/**
+	 * Create a map of the merged Kafka admin properties from the Kafka Spring properties.
+	 * @param properties the Kafka Spring properties
+	 * @return a Map containing all Kafka admin properties
+	 */
+	protected abstract Map<String, Object> getMergedAdminProperties(T properties);
+
+	/**
+	 * Get the raw {@link Map} object from the Kafka Spring properties that stores admin-specific properties.
+	 * @param properties the Kafka Spring properties
+	 * @return the map from Kafka properties storing admin-specific properties
+	 */
+	protected abstract Map<String, String> getRawAdminProperties(T properties);
+
+	protected abstract boolean needsPostProcess(Object bean);
+
+	protected abstract Logger getLogger();
+
+	/**
+	 * Process Kafka Spring properties for any customized operations.
+	 * @param properties the Kafka Spring properties
+	 */
+	protected void customize(T properties) {
+	}
+
+
+	protected void clearAzureProperties(Map<String, String> properties) {
+		AzureKafkaPropertiesUtils.AzureKafkaPasswordlessPropertiesMapping.getPropertyKeys()
+			.forEach(properties::remove);
+	}
+
+	/**
+	 * This method executes two operations:
+	 * <p>
+	 * 	1. When this configuration meets Azure Kafka passwordless startup requirements, convert all Azure properties
+	 * 	in Kafka to {@link Jaas}, and configure the JAAS configuration back to Kafka.
+	 * </p>
+	 * <p>
+	 * 	2. Clear any Azure properties in Kafka properties.
+	 * </p>
+	 * @param mergedProperties the merged Kafka properties which can contain Azure properties to resolve JAAS from
+	 * @param rawPropertiesMap the raw Kafka properties Map to configure JAAS to and remove Azure Properties from
+	 */
+	private void replaceAzurePropertiesWithJaas(Map<String, Object> mergedProperties, Map<String, String> rawPropertiesMap) {
+		resolveJaasForAzure(mergedProperties)
+			.ifPresent(jaas -> {
+				configJaasToKafka(rawPropertiesMap, jaas);
+				configureKafkaUserAgent();
+			});
+		clearAzureProperties(rawPropertiesMap);
+	}
+
+	private Optional<Jaas> resolveJaasForAzure(Map<String, Object> mergedProperties) {
```

Comment:
I like the usage of `Optional` and I think this is a very good example of when and when only to use `Optional` api